sentinel.go

```diff
@@ -0,0 +1,92 @@
+package sentinel
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+func New(ctx context.Context, config Config) *sentinel {
+	c := make(chan bool)
+	return &sentinel{
+		Config: config,
+		active: false,
+		ticker: time.NewTicker(config.Duration),
+		stop:   make(chan bool, 1),
+		c:      c,
+		Done:   c,
+	}
+}
+
+func (s *sentinel) Start() error {
```

Comment:
Would be useful to allow start to be called after Stop as a sort of "Resume" feature. Shouldn't be too difficult to rework for that functionality